core/types/gen_header_json.go

```diff
@@ -243,3 +224,130 @@ func (t *Termini) UnmarshalJSON(input []byte) error {
 	t.SetSubTermini(dec.SubTermini)
 	return nil
 }
+
+func (wh *WorkObjectHeader) MarshalJSON() ([]byte, error) {
+	var enc struct {
+		HeaderHash common.Hash   `json:"headerHash" gencoden:"required"`
+		ParentHash common.Hash   `json:"parentHash" gencoden:"required"`
+		Number     *hexutil.Big  `json:"number" gencoden:"required"`
+		Difficulty *hexutil.Big  `json:"difficulty" gencoden:"required"`
+		TxHash     common.Hash   `json:"txHash" gencoden:"required"`
+		Location   hexutil.Bytes `json:"location" gencoden:"required"`
+		MixHash    common.Hash   `json:"mixHash" gencoden:"required"`
+		Nonce      BlockNonce    `json:"nonce" gencoden:"required"`
+	}
+
+	enc.HeaderHash = wh.HeaderHash()
+	enc.Difficulty = (*hexutil.Big)(wh.Difficulty())
+	enc.Number = (*hexutil.Big)(wh.Number())
+	enc.TxHash = wh.TxHash()
+	enc.Location = hexutil.Bytes(wh.Location())
+	enc.MixHash = wh.MixHash()
+	enc.Nonce = wh.Nonce()
+
+	raw, err := json.Marshal(&enc)
+	return raw, err
+}
+
+func (wh *WorkObjectHeader) UnmarshalJSON(input []byte) error {
+	var dec struct {
+		HeaderHash common.Hash     `json:"headerHash" gencoden:"required"`
+		ParentHash common.Hash     `json:"parentHash" gencoden:"required"`
+		Number     *hexutil.Big    `json:"number" gencoden:"required"`
+		Difficulty *hexutil.Big    `json:"difficulty" gencoden:"required"`
+		TxHash     common.Hash     `json:"txHash" gencoden:"required"`
+		Location   common.Location `json:"location" gencoden:"required"`
```

Comment:
this has to be hexutil bytes to decode properly since its a slice, wont work otherwise

Reply:
fixed